Make IndentBlock work regardless of editor plugins order.

jodator opened the pull request:
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Make IndentBlock work regardless of editor plugins order. Closes ckeditor/ckeditor5#2359.

--- 

Checking schema for registered elements should be done
in afterInit call to prevent plugin ordering related bugs.

jodator committed:
Make IndentBlock work regardless of editor plugins order.

Checking schema for registered elements should be done
in afterInit call to prevent plugin ordering related bugs.

Reinmar reviewed (changes requested):
The minimal amount of code should be moved to `afterInit()`. As much as possible should be done inside `init()` to make those things available to other plugins.

jodator committed:
Split indent block feature initialization to init and afterInit.

The schema extending rules must be done in afterInit and
all core indent block behavior should be done during
the init phase.

jodator commented:
I think that now it makes more sense.

oleq commented on src/indentblock.js:
```diff
@@ -82,9 +72,20 @@ export default class IndentBlock extends Plugin {
 		const outdentCommand = editor.commands.get( 'outdent' );
 
+		// Enable block indentation by default in paragraph and default headings.
+		const knownElements = [ 'paragraph', 'heading1', 'heading2', 'heading3', 'heading4', 'heading5', 'heading6' ];
```
I was wondering since we're after the initialization phase, maybe we could figure out some better way to learn about model element names? For instance, the heading feature is [configurable](https://ckeditor.com/docs/ckeditor5/latest/api/module_heading_heading-HeadingConfig.html#member-options) and these may not be necessarily the true names of elements. 

Maybe we can fish them out of the heading config? No need to use `schema.isRegistered( elementName )` then.

jodator replied:
Well, we could but I don't know if this is worth the hassle:

```js
const knowElements = [ 
    'paragraph', // we must work withouth Heading feature
    ...( editor.config.get( 'heading.options' ) || [] )
        .map( option => option.model )
        .filter( name => name !== 'pargraph' ) /// and we don't want double entries
].forEach( name => schema.extend( name. { allowAttributes: 'blockIndent' } ) );
```

jodator replied:
For me both are equally fine.

oleq replied:
I think the following should do the trick

```js
new Set( [ 
	'paragraph',
	...editor.config.get( 'heading.options' ).map( ( { model } ) => model )
] ).forEach( name => schema.extend( name, { allowAttributes: 'blockIndent' } ) )
```

`Set` ensures unique values, and I guess some `heading.options` are always defined (`editor.config.define` in `HeadingEditing`).

oleq replied:
But this is not the scope of this PR I guess. I created a separate issue https://github.com/ckeditor/ckeditor5/issues/5565.

jodator replied:
OH, forgot about the `Set` :D Now it looks better and we could do that.

oleq committed:
Tests: Added a link to the issue. Improved language in the test description.
